Microsoft.HealthVault.Fhir/Transformers/AllergyIntoleranceToHealthVault.cs

```diff
@@ -0,0 +1,157 @@
+            }
+
+            var allergy = allergyIntolerance.ToThingBase<Allergy>();
+
+            string allergenType = string.Empty;
+            var allergyExtension = allergyIntolerance.GetExtension(HealthVaultExtensions.Allergy);
+            if (allergyExtension != null)
+            {
+                allergenType = allergyExtension.GetStringExtension(HealthVaultExtensions.AllergenType);
+                allergy.AllergenCode = allergyExtension.GetExtensionValue<CodeableConcept>(HealthVaultExtensions.AllergenCode)?.ToCodableValue();
+                allergy.Treatment = allergyExtension.GetExtensionValue<CodeableConcept>(HealthVaultExtensions.AllergyTreatement)?.ToCodableValue();
+
+            }
+
+            foreach (var code in allergyIntolerance.Code.Coding)
+            {
+                if (HealthVaultVocabularies.SystemContainsHealthVaultUrl(code.System))
+                {
+                    allergy.Name = allergyIntolerance.Code.ToCodableValue();
+                }
+                else
+                {
+                    allergy.SetAllergyName(
+                        code.Display,
+                        code.Code,
+                        HealthVaultVocabularies.Fhir,
+                        code.System,
+                        code.Version);
+                }
+            }
+
+            if (allergyIntolerance.Reaction != null && allergyIntolerance.Reaction.Count > 0)
+            {              
+                foreach (var code in allergyIntolerance.Reaction.FirstOrDefault().Manifestation.FirstOrDefault().Coding)
+                {
+                    if (HealthVaultVocabularies.SystemContainsHealthVaultUrl(code.System))
+                    {
+                        allergy.Reaction = allergyIntolerance.Reaction.FirstOrDefault().Manifestation.FirstOrDefault().ToCodableValue();
+                    }
+                    else
+                    {
+                        allergy.SetAllergyReaction(
+                            code.Display,
+                            code.Code,
+                            HealthVaultVocabularies.Fhir,
+                            code.System,
+                            code.Version);
+                    }
+                }
+            }
+
+            if (allergyIntolerance.Onset != null)
+            {
+                allergy.FirstObserved = allergyIntolerance.Onset.ToAproximateDateTime();
+            }
+
+            if (allergyIntolerance.Category != null && allergyIntolerance.Category.Count() > 0)
+            {
+                allergy.AllergenType = new CodableValue(allergyIntolerance.Category.FirstOrDefault().Value.ToString())
+                {
+                    new CodedValue(allergyIntolerance.Category.FirstOrDefault().Value.ToString(), FhirCategories.HL7Allergy, HealthVaultVocabularies.Fhir, "")
+                };
+            }
+            else if (!string.IsNullOrWhiteSpace(allergenType))
```

Comment:
Would it make more sense to optimistically use the allergenType first instead of the category since it will only be set if the the origination came from health vault?